pkg/generator/secret_template_reconciler.go

```diff
@@ -283,7 +284,12 @@ func evaluateTemplate(template *sg2v1alpha1.JSONPathTemplate, values map[string]
 	}
 
 	// Template Secret StringData
-	stringData, err := evaluate(template.StringData, values)
+	decodedValues, err := decodeSecrets(values)
```

Comment:
This is a pre-processing step which ends up having to iterate all the values even if you never specify a SecretData field.

Is there a way to fold this into our existing `evaluate` function or a new function `evaluateStringData` which could delegate to the existing evaluateBytes or similar for most of the work.